### PR TITLE
Ensure the ServiceAccount/Role/ClusterRole resources are created in order

### LINKED ISSUE #32640: Openshift application has no access to secrets
### Describe the bug

I have secrets already created. I deploy an application to OpenShift for the first time in this namespace / project. The application cannot access the secrets I created before.

I'm using the kubernetes-config extension. Seems to be another service account issue, not applied at the right time.

If I delete the deployment config and re-deploy it, it works. 

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Sgitario**
Fix https://github.com/quarkusio/quarkus/issues/32640 
Partially reverts https://github.com/quarkusio/quarkus/pull/32208

### CODE REVIEW

**Comment by iocanel** [comment]
Given that we allow users to bring their own resources, it's possible to end up with dependency chains that are contains 3 or more items (e.g. Namespace -> ConfigMap -> Deployment).

I suggest we just order resources in a way similar to: https://github.com/superbrothers/ksort/blob/af6d8f61cdc596ef660c1ffd64c6ca8604b0cb91/kind_sorter.go#L29

**Comment by Sgitario** [comment]
> Given that we allow users to bring their own resources, it's possible to end up with dependency chains that are contains 3 or more items (e.g. Namespace -> ConfigMap -> Deployment).
> 
> I suggest we just order resources in a way similar to: https://github.com/superbrothers/ksort/blob/af6d8f61cdc596ef660c1ffd64c6ca8604b0cb91/kind_sorter.go#L29

+100 
PR updated with this list.

**Comment by maxandersen** [comment]
marking as backport as it result in app deployment error.

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/QuarkusFileWriter.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            "CustomResourceDefinition",
+            CLUSTER_ROLE,
+            "ClusterRoleList",
+            CLUSTER_ROLE_BINDING,
+            "ClusterRoleBindingList",
+            ROLE,
+            "RoleList",
+            ROLE_BINDING,
+            "RoleBindingList",
+            "Service",
```
Comment (iocanel) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/QuarkusFileWriter.java:51]:
```suggestion
            "Service",
            "ImageStream",
            "BuildConfig",
```

**Review by iocanel — CHANGES_REQUESTED** [review]
Looks good, but we should also include `ImageStream` and `BuildConfig` resources.

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/QuarkusFileWriter.java (hunk 1 above)
Comment (Sgitario) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/QuarkusFileWriter.java:51]: Done

**Comment by Sgitario** [comment]
PR updated by adding the ImageStream and BuildConfig resources.

**Comment by gsmet** [comment]
@iocanel I let you approve and merge if you're happy with the changes.

**Review by iocanel — APPROVED** [review]
lgtm

**Pull request merged**